sensor_test.py

```diff
@@ -202,5 +201,64 @@ def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
     sensor_info_after_update = get_sensor_info()
     current_firmware_version = sensor_info_after_update.firmware_version
     assert (
-        current_firmware_version == max_firmware_version
+            current_firmware_version == max_firmware_version
     ), "Sensor firmware version not max"
+
+
+def test_set_empty_sensor_name(get_sensor_info, set_sensor_name):
+    """
+    Test Steps:
+    1. Get original sensor name.
+    2. Set sensor name to an empty string.
+    3. Validate that sensor responds with an error.
+    4. Get current sensor name.
+    5. Validate that sensor name didn't change.
+    """
+
+    log.info("1. Get original sensor name")
+    sensor_info = get_sensor_info()
+    original_name = sensor_info.name
+
+    log.info("2. Set sensor name to an empty string")
+    empty_name = ""
+    response_to_empty_name = set_sensor_name(empty_name)
+
+    log.info("3. Validate that sensor responds with an error")
+    assert response_to_empty_name == "error", "Sensor didn't respond with an error"
+
+    log.info("4. Get current sensor name")
+    sensor_info_after_empty_name = get_sensor_info()
+    current_name = sensor_info_after_empty_name.name
+
+    log.info("5. Validate that sensor name didn't change")
+    assert current_name == original_name, "Sensor name changed when it shouldn't have"
+
+
+def test_set_invalid_sensor_reading_interval(get_sensor_info, set_sensor_reading_interval):
+    """
+    Test Steps:
+        1. Get original sensor reading interval.
+        2. Set interval to < 1
+        3. Validate that sensor responds with an error.
+        4. Get current sensor reading interval.
+        5. Validate that sensor reading interval didn't change.
+    """
+
+    log.info("1. Get original sensor reading interval")
+    sensor_info = get_sensor_info()
+    original_interval = sensor_info.reading_interval
+
+    log.info("2. Set interval to < 1")
+    invalid_interval = 0
+    response_to_invalid_interval = set_sensor_reading_interval(invalid_interval)
+
+    log.info("3. Validate that sensor responds with an error")
+    assert response_to_invalid_interval == "error", "Sensor didn't respond with an error"
```

Comment:
Directly relating to the first comment: here's how the check would look like if you keep `_make_valid_request()` unchanged:
```
assert response_to_empty_name =={}, "Sensor didn't respond with an error"
```